### ISSUE TITLE
Socket accept() should happen outside __init__()

### ISSUE THREAD

**Issue by tcalmant** [tamentis/rpdb#4]
Hi,

The self.skt.accept() method should be called from a separate method (start(), accept() or similar), in order to let the `__init__()` return immediately after the creation of the servlet.
An "immediate_accept" argument could be added to `__init__()` to support both ways to work.

This would allow to log the actual listening port (i.e. if 0 is given as a port) before accepting connections.
It would also allow to close the socket even if no client has arrived yet.

**Comment by shaleh** [comment]
But if you did that the Pdb object would not have its ``__init__`` called until the method containing ``accept()`` is called.

**Comment by tcalmant** [comment]
Yes, but it is not a problem as long as it is documented and not the default behaviour.

Another solution could be to accept a callback method in parameter of ``__init__()``, which would be called after the call ``bind()`` and before the call to ``accept()``.

The goal is only to have access to the listening port of rpdb before it blocks the execution, waiting for a client.

**Comment by shaleh** [comment]
Then perhaps the right thing to do is to refactor so that the Rpdb object contains a Pdb object instead of derives from it. This way the Pdb object could be instantiated closer to when it is needed.
